Source/WebCore/dom/Document.cpp

```diff
@@ -3578,8 +3578,11 @@ const URL& Document::urlForBindings() const
         if (m_url.url().isEmpty() || !loader() || !isTopDocument() || !frame())
             return false;
 
-        auto* topDocumentLoader = topDocument().loader();
-        if (!topDocumentLoader || !topDocumentLoader->networkConnectionIntegrityPolicy().contains(NetworkConnectionIntegrity::Enabled))
+        auto* targetLoader = topDocument().loader();
+        if (targetLoader && !targetLoader->request().url().hasSpecialScheme() && url().hasSpecialScheme())
```

Comment:
The top document loader should probably check hasSpecialScheme, but this document's URL should probably just check protocolIsInHTTPFamily.  I don't think we want to do this for file URLs loaded in iframes, for example.

Also, I think looking at a new document loader needs a linked-on-or-after check.  Otherwise, existing applications that expect the top document's document loader to be consulted without giving the same policies to iframes will be broken.